libs/permit-utils/src/lib/generatePermitHook.ts

```diff
@@ -6,28 +6,26 @@ import { buildDaiLikePermitCallData, buildEip2162PermitCallData } from '../utils
 import { getPermitDeadline } from '../utils/getPermitDeadline'
 import { isSupportedPermitInfo } from '../utils/isSupportedPermitInfo'
 
-const REQUESTS_CACHE: { [permitKey: string]: Promise<PermitHookData> } = {}
+const REQUESTS_CACHE: { [permitKey: string]: Promise<PermitHookData | undefined> } = {}
 
-export async function generatePermitHook(params: PermitHookParams): Promise<PermitHookData> {
+export async function generatePermitHook(params: PermitHookParams): Promise<PermitHookData | undefined> {
   const permitKey = getCacheKey(params)
 
   const cachedRequest = REQUESTS_CACHE[permitKey]
 
   if (cachedRequest) {
-    try {
-      return await cachedRequest
-    } catch (e) {
-      console.debug(`[generatePermitHookWith] cached request failed`, e)
-      delete REQUESTS_CACHE[permitKey]
-    }
+    return await cachedRequest
```

Comment:
Exception was caught here only if there was a cached request, but it was not handled when not cached